content/courses/intro-to-embedded-sys-prog/chapters/low_level_programming.rst

```diff
@@ -579,41 +657,919 @@ sending values over the wire, the code should query the size of the
+
+We said earlier that there are only a small number of reasons to specify 
+:ada:`'Size` for a type. We can mention one of them now. Setting 'Size 
+can be useful to give the minimum number of bits to use for a component 
+of a packed composite type, that is, within either a record type or an 
+array type that is explicitly packed via the aspect or pragma Pack. It 
+says that the compiler, when giving its best effort, shouldn't compress 
+components of the type any smaller than the number of bits specified. 
+No, it isn't earth-shattering, but other uses are more valuable, to be 
+discussed soon. 
+
+One thing we will leave unaddressed (pun intended) is the question of 
+bit ordering and byte ordering within our record layouts. In other 
+words, the "endian-ness". That's a subject beyond the scope of this 
+course. Suffice it to say that GNAT provides a way to specify record 
+layouts that are independent of the endian-ness of the machine, within 
+some implementation-oriented limits. That's obviously useful when the 
+code might be compiled for a different ISA in the future. On the other 
+hand, if your code is specifically for a single ISA, e.g. Arm, even if 
+different boards and hardware vendors are involved, there's no need to 
+be independent of the endian-ness. It will always be the same in that 
+case. (Those are "famous last words" though.) 
+
+Although specifying record type layouts and alignments are perhaps the 
+most common representation characteristics expressed, there are a couple 
+of other useful cases. Both involve storage allocation.
+
+One useful scenario concerns tasking. We can specify the number of 
+storage elements reserved for the execution of a task object, or all 
+objects of a task type. You use the Storage_Size aspect to do so: 
+
+.. code-block:: ada
+
+   task Servo with
+     Storage_Size => 1 * 1024,
+     ...
+     
+Or the corresponding pragma:
+
+.. code-block:: ada
+
+   task Servo is
+      pragma Storage_Size (1 * 1024);
+   end Servo;     
+
+The aspect seems textually cleaner and lighter unless you have task 
+entries to declare as well. In that case the line for the pragma 
+wouldn't add all that much. That's a matter of personal aesthetics 
+anyway. 
+
+The specified number of storage elements includes the size of 
+the task's stack (GNAT does have one, per task). The language does not 
+specify whether or not it includes other storage associated with the 
+task used for implementing and managing the task execution. With GNAT, 
+the extent of the primary stack size is the value returned, ignoring any 
+other storage used internally in the run-time library for managing the 
+task. 
+
+The GNAT run-time library allocates a default stack amount to each task, 
+with different defaults depending on the underlying O.S., or lack 
+thereof, and the target. You need to read the documentation to find the 
+actual amount, or, with GNAT, read the code (or just ask for help via 
+your GNAT account number). 
```

Comment:
I believe we shouldn't mention the "account number" in this book.

Reply:
Thanks!